src/dsl/lb.rs

```diff
@@ -0,0 +1,199 @@
+use std::{
+    collections::HashMap,
+    fmt::Debug,
+    sync::{Arc, Mutex},
+};
+
+use crate::{
+    ast::{Expr, Lookup},
+    util::{uuid, UUID},
+};
+
+use super::{cb::Constraint, StepTypeSetupContext};
+
+pub trait LookupBuilder<F> {
+    fn build(self, ctx: &StepTypeSetupContext<F>) -> Lookup<F>;
+}
+
+/// A helper struct for building lookup tables.
+pub struct InPlaceLookupBuilder<F> {
+    lookup: Lookup<F>,
+}
+
+impl<F> LookupBuilder<F> for InPlaceLookupBuilder<F> {
+    fn build(self, _: &StepTypeSetupContext<F>) -> Lookup<F> {
+        self.lookup
+    }
+}
+
+impl<F> Default for InPlaceLookupBuilder<F> {
+    fn default() -> Self {
+        InPlaceLookupBuilder {
+            lookup: Lookup::default(),
+        }
+    }
+}
+
+impl<F: Debug + Clone> InPlaceLookupBuilder<F> {
+    /// Adds a source column-lookup column pair to the lookup table. Because the function returns a
+    /// mutable reference to the `LookupBuilder<F>`, it can an chain multiple `add` and `enable`
+    /// function calls to build the lookup table. Requires calling `lookup` to create an empty
+    /// `LookupBuilder` instance at the very front.
+    pub fn add<C: Into<Constraint<F>>, E: Into<Expr<F>>>(
+        mut self,
+        constraint: C,
+        expression: E,
+    ) -> Self {
+        let constraint = constraint.into();
+        self.lookup
+            .add(constraint.annotation, constraint.expr, expression.into());
+        self
+    }
+
+    /// Adds a selector column specific to the lookup table. Because the function returns a mutable
+    /// reference to the `LookupBuilder<F>`, it can an chain multiple `add` and `enable` function
+    /// calls to build the lookup table. Requires calling `lookup` to create an
+    /// empty `LookupBuilder` instance at the very front.
+    pub fn enable<C: Into<Constraint<F>>>(mut self, enable: C) -> Self {
+        let enable = enable.into();
+        self.lookup.enable(enable.annotation, enable.expr);
+        self
+    }
+}
```

Comment:
This looks like the deprecated lookup builder. It doesn't implement the `LookupBuilder` trait which `add_lookup` in dsl requires. It also doesn't have a public `new` method so it cannot really be created. Maybe we can just delete it?